query/src/main/java/io/keen/client/java/KeenQueryRequest.java

```diff
@@ -0,0 +1,27 @@
+package io.keen.client.java;
+
+import io.keen.client.java.exceptions.KeenQueryClientException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Interface to be implemented by a query request
+ *
+ * @author baumatron
+ */
+abstract class KeenQueryRequest {
```

Comment:
Question: Why use an `abstract class` here instead of an actual `interface`? Is it just because of the default implementation in `getGroupByParams`? Doesn't necessarily seem worth it to me, but it's not a strong opinion.